File: backend/src/main/java/com/dollyanddot/kingmaker/domain/routine/dto/MemberRoutineDto.java
package com.dollyanddot.kingmaker.domain.routine.dto;

import com.dollyanddot.kingmaker.domain.routine.domain.MemberRoutine;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class MemberRoutineDto {

  private Long memberRoutineId;
  private RoutineDto routine;
  private boolean achievedYn;
  private int monsterCd;

  public static MemberRoutineDto from(MemberRoutine memberRoutine){
    return MemberRoutineDto.builder()
        .memberRoutineId(memberRoutine.getId())
        .routine(RoutineDto.from(memberRoutine.getRoutine()))
        .achievedYn(memberRoutine.isAchievedYn())
        .monsterCd(memberRoutine.getMonsterCd())
        .build();
  }
}

File: frontend/kingmaker/lib/dto/member_routine_dto.dart
import 'package:kingmaker/dto/routine_dto.dart';

class MemberRoutineDto{
  int memberRoutineId;
  RoutineDto routine;
  bool achievedYn;
  bool importantYn;

  MemberRoutineDto({
    required this.memberRoutineId,
    required this.routine,
    required this.achievedYn,
    required this.importantYn,
  });

  factory MemberRoutineDto.fromJson(Map<String, dynamic> json) {
    return MemberRoutineDto(
        memberRoutineId: json['memberRoutineId'],
        routine: RoutineDto.fromJson(json['routine']),
        achievedYn: json['achievedYn']==1? true : false,
        importantYn: json['importantYn']==1?true: false

    );


  }
}